WikiWatershed/model-my-watershed
# BiG-CZ: CUAHSI Variable Values using Ulmo and WaterML

## Overview

Fetches values from CUAHSI / WDC using Ulmo for each variable at a site. We're currently only showing the latest value in the table, with charts coming in a follow-up PR.

Connects #2243 
Connects #2238 

### Demo

![2017-10-05 12 54 37](https://user-images.githubusercontent.com/1430060/31239884-b2a04184-a9cc-11e7-9ec1-a791f400c1eb.gif)

### Notes

This PR adds two endpoints: `/details` and `/values`. Currently we are only using `/values` in the UI, but may use `/details` in the future. There is a dummy placeholder for the charts, but no values.

### Known Issues

The popover for the headline for WDC Details View doesn't work.

## Testing Instructions

 * Check out this branch and reprovision `app` and `worker`. Run the `bundle` script.
 * Go to [:8000/?bigcz](http://localhost:8000/?bigcz), pick a shape and search for something
 * Go to the WDC tab. Click on an item to go to the details view.
 * Ensure you see a spinner fetching the values.
 * As values succeed, ensure you see them being filled in the table one by one.
 * Ensure if fetching succeeds for all values, the "Last Updated" value is updated to be more recent. This happens because the API call used for the initial load has older values than the new ones that come via this second API.
 * Ensure that if any fetch fails, there is an error icon where the spinner goes.
